src/components/modal/UserModalSelect.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import Users from '../../screens/users/list/Users';
+import BaseProps from '../../types/BaseProps';
+import ListItem from '../../types/ListItem';
+import User from '../../types/User';
+import Utils from '../../utils/Utils';
+import { ItemSelectionMode } from '../list/ItemsList';
+import ModalSelect from './ModalSelect';
+
+export interface Props extends BaseProps {
+  onUserSelected: (items: User[]) => void;
+  defaultUser?: User;
+  selectionMode: ItemSelectionMode;
```

Comment:
If only one user why selection mode prop?
It's de facto single select?
If you pass the selectionMode you assume that you always have many users -> so all are arrays.
If this is what you want rename the class UsersModalSelect, onUsersSelected...
Otherwise remove selectionMode and if multi select is needed a new class will be created: UsersModalSelect
Alors ?

Reply:
I did this in case we want to have a multi select users modal, we just have one prop to change

Reply:
There is no default selected item.
I would use the same modal for handling single or multiple selection, so everything as to be plural.
If single select: you select one line, the modal closes and sends the result back to the caller in an Array with one record.
If multi select: you select multiple users and click on a button to validate your choice (button below the header?) and you return the selected users in an array.

Reply:
I remember you wanted to put the current user in the field. This is what I call the defaultUser

Reply:
Hum okay for this variable, maybe selectedUsers would be a better naming which contains a string like in Angular:
<firstName> <Name> (+n) 
For instance:
Serge FABIANO (+2)
-> Meaning The first selected user first and last name and you selected 2 more users.